### PR TITLE
Add support for + and - operators in immutable maps

### PR DESCRIPTION
**Pull request by K0zka**
Fixes: KT-6594
Bug-Url: https://youtrack.jetbrains.com/issue/KT-6594
Signed-off-by: Laszlo Hornyak <laszlo.hornyak@gmail.com>

### CODE REVIEW

File: libraries/stdlib/src/kotlin/collections/Maps.kt [hunk 1]
```diff
@@ -264,3 +264,35 @@ public fun <K, V> Iterable<Pair<K, V>>.toMap(): Map<K, V> {
...
+
+/**
+ * Create a new read-only map by removing a key
+ */
+public fun <K, V> Map<K, V>.minus(key: K): Map<K, V> {
+    return this.filterKeys { key != it }
+}
+
+/**
+ * Create a new read-only map by removing a list of keys
```
Comment (K0zka) [inline libraries/stdlib/src/kotlin/collections/Maps.kt]: > list of

oops

**Comment by orangy** [comment]
Working on integrating the pull request. It's not possible right away, because it causes ambiguity between plus() and plusAssign() on mutable maps. We will update compiler to resolve ambiguity and then merge.

**Comment by K0zka** [comment]
Thank you, Ilya!

**Comment by orangy** [comment]
Fixed issues, added overloads, merged manually.

**Pull request closed without merging**